### PR TITLE
Documentation: Polish "refresh_pattern percent" description

### PR DESCRIPTION
**Pull request by rousskov**
Authored-by: Amos Jeffries <squid3@treenet.co.nz>

The original text contained a "last modification age" typo and was
misinterpreted by some admins as if Squid applied the configured percent
to the current object age.

The text also did not make it clear that the percent-based heuristic is
effectively only applied to responses with a Last-Modified header (in
addition to the usual "without an explicit expiry time" precondition).

### CODE REVIEW

File: src/cf.data.pre [hunk 1]
```diff
@@ -6454,9 +6454,16 @@ DOC_START
...
+	'Percent' is used to compute the expiration time of responses with a
+	Last-Modified header and without an explicit expiry time: When Squid
+	caches (or updates) such a response, the expiration time is effectively
+	set to the given percentage of the time passed since last modification.
+	For example, if such an object was cached 1000 minutes after its
+	Last-Modified time and percent is set to 20%, then the object will be
+	considered fresh for the first 200 minutes (1000*0.20=200) it spends in
+	the cache. The lm-factor term used in the algorithm sketch below is object
+	age divided by stable-time, where stable-time is the time between the last
+	object modification and the moment the object was cached (or updated).
```
Comment (rousskov) [inline src/cf.data.pre:6467]: I am not sure the whole algorithm sketch below and its `lm-factor` line in particular can be correctly interpreted by most admins. To avoid broadening this PR scope to the whole refresh_pattern documentation rewrite (which I cannot volunteer for right now), I decided to leave that algorithm sketch untouched in this PR and just document what its `lm-factor` term is. 

Ideally, the algorithm sketch should be moved much higher (and described a bit better) because many admins do not realize that it determines critical heuristics application order (and some probably stop reading after hitting the long list of override options/flags!). Again, I decided not to touch that for now.

I will quote that algorithm sketch here for convenience:

```
	Basically a cached object is:

		FRESH if expire > now, else STALE
		STALE if age > max
		FRESH if lm-factor < percent, else STALE
		FRESH if age < min
		else STALE
```

**Review by rousskov — COMMENTED** [review]
I think the proposed text is better than the current official version, but I am sure this documentation can be significantly improved further, and I do not insist on any specific wording. To save time, please edit the text directly or suggest very specific wording replacements. Thank you.

**Review by kinkie — APPROVED** [review]
I'm not sure the extra clarification is needed for most admins, but it won't hurt, so let's ship it

**Comment by rousskov** [comment]
> I'm not sure the extra clarification is needed for most admins, but it won't hurt, so let's ship it

I am surprised be the focus on questioning the need for _clarification_ when the existing text is simply buggy, but I am glad we agree that a better text should go in (in some shape of form).

File: src/cf.data.pre [hunk 2]
```diff
@@ -6454,9 +6454,16 @@ DOC_START
...
```
[the remark this answers is not in the document]
Comment (rousskov) [reply src/cf.data.pre]: IMO, too many readers will think that `Date` in the above suggestion is the current transaction time rather than the Date header of the previously cached response.

I am also concerned that folks will misinterpret this as Squid actually adding the `Cache-control: max-age` header to the response.

Finally, the above version does not fix the `lm-factor` problem.

Nevertheless, I have merged this suggestion and adjusted PR description accordingly (please adjust further as needed). When you are happy with the result, please clear this PR for merging.

**Review by yadij — APPROVED** [review]
